### PR TITLE
Bug fix/validate transaction collection

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [ ] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as *graph transaction tests*.

https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/6196/

### CODE REVIEW

File: arangod/StorageEngine/TransactionState.cpp [hunk 1]
```diff
@@ -202,6 +202,24 @@ void TransactionState::allCollections(                     // iterate
     }
   }
 }
+  
+/// @brief whether or not the collection is used in the transaction
+bool TransactionState::containsCollection(TRI_voc_cid_t cid, AccessMode::Type accessType) {
```
Comment (graetzer) [inline arangod/StorageEngine/TransactionState.cpp]: I think in a running transaction we should be able to check the transaction name

File: arangod/RestHandler/RestVocbaseBaseHandler.cpp [hunk 2]
```diff
@@ -584,6 +585,18 @@ std::unique_ptr<SingleCollectionTransaction> RestVocbaseBaseHandler::createTrans
       LOG_TOPIC("e94ea", DEBUG, Logger::TRANSACTIONS) << "Transaction with id '" << tid << "' not found";
       THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_TRANSACTION_NOT_FOUND, std::string("transaction '") + std::to_string(tid) + "' not found");
     }
+    auto* state = ctx->getParentTransaction();
+    TRI_ASSERT(state != nullptr);
+    TRI_voc_cid_t cid = 0;
+    if (state->isCoordinator()) {
```
Comment (graetzer) [inline arangod/RestHandler/RestVocbaseBaseHandler.cpp]: I think we should just check the name against the name string or the ID

File: arangod/StorageEngine/TransactionState.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/StorageEngine/TransactionState.cpp]: I have not the slightest idea what a transaction name is.

File: arangod/RestHandler/RestVocbaseBaseHandler.cpp (hunk 2 above)
Comment (jsteemann) [reply arangod/RestHandler/RestVocbaseBaseHandler.cpp]: I don't understand. Can you elaborate?

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/6244/

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/6250/

**Pull request merged**